### ISSUE TITLE
Conformance testing

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/unicode-transforms#38]
Need to implement unciode conformance testing. See  https://www.unicode.org/reports/tr15/tr15-18.html#Conformance%20Testing .

**Comment by harendra-kumar** [comment]
We use the unicode normalization test suite for testing, but we need to check if there are any more resources/ways to test so that testing is robust and complete. Does standard specify more ways to test?

**Comment by Bodigrim** [comment]
One could possibly modify `QuickCheckUtils.hs` to generate "more interesting" strings, with increased density of combining/decomposable characters.

**Comment by harendra-kumar** [comment]
Yes, and we can do the same for testing against ICU. We have a testsuite that normalizes random strings using unicode-transforms and ICU both and then compares the results. There as well we need to check if we are using interesting strings.
